core/Translation/Translator.php

```diff
@@ -185,15 +188,36 @@ public function setCurrentLanguage($language)
     }
 
     /**
-     * @return string The default configured language.
+     * @return string The default language.
      */
     public function getDefaultLanguage()
```

Comment:
I think this method does no longer return the default language but the actual configured language. We should maybe rename the method which we cannot do before Piwik 3.0.0 to keep backwards compatibility. Therefore it might be better to add a new method. 

There are some other problems though. The translator class is in `core` but accesses code from plugins which is not really good since the core should not know anything about the plugins. Also it would be nice if this class wouldn't contain too much logic like deciding from where to read the language.

Maybe we could move that code to a method in `LanguagesManager`?